iree/compiler/Dialect/VM/IR/test/assignment_folding.mlir

```diff
@@ -58,3 +58,42 @@ vm.module @select_ref_folds {
     vm.return %0 : !vm.ref<?>
   }
 }
+
+// -----
+
+// CHECK-LABEL: @switch_i32_folds
+vm.module @switch_i32_folds {
+  // CHECK-LABEL: @switch_i32_nop
+  vm.func @switch_i32_nop(%arg0 : i32) -> i32 {
+    %c5 = vm.const.i32 5 : i32
+    // CHECK: vm.return %c5 : i32
+    %0 = vm.switch.i32 [](%arg0) else %c5 : i32
+    vm.return %0 : i32
+  }
+
+  // CHECK-LABEL: @switch_i32_identical
+  vm.func @switch_i32_identical(%arg0 : i32) -> i32 {
+    %c100 = vm.const.i32 100 : i32
+    // CHECK: vm.return %c100 : i32
+    %0 = vm.switch.i32 [%c100, %c100, %c100](%arg0) else %c100 : i32
+    vm.return %0 : i32
+  }
+
+  // CHECK-LABEL: @switch_i32_constant_index
+  vm.func @switch_i32_constant_index() -> (i32, i32, i32, i32) {
+    %c0 = vm.const.i32 0 : i32
+    %c1 = vm.const.i32 1 : i32
+    %c2 = vm.const.i32 2 : i32
+    %c3 = vm.const.i32 3 : i32
+    %c100 = vm.const.i32 100 : i32
+    %c200 = vm.const.i32 200 : i32
+    %c300 = vm.const.i32 300 : i32
+    %c400 = vm.const.i32 400 : i32
+    // CHECK: vm.return %c100, %c200, %c300, %c400 : i32, i32, i32, i32
+    %0 = vm.switch.i32 [%c100, %c200, %c300](%c0) else %c400 : i32
```

Comment:
Nit: I find this syntax hard to parse. Reversing it makes more sense to me:
(%c0)[...] else %...

Reply:
done